ci-operator/step-registry/mirror-images/by-oc-adm-in-bastion/mirror-images-by-oc-adm-in-bastion-commands.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+mirror_output="${SHARED_DIR}/mirror_output"
+new_pull_secret="${SHARED_DIR}/new_pull_secret"
+install_config_icsp_patch="${SHARED_DIR}/install-config-icsp.yaml.patch"
+icsp_file="${SHARED_DIR}/local_registry_icsp_file.yaml"
+
+# private mirror registry host
+# <public_dns>:<port>
+MIRROR_REGISTRY_HOST=`head -n 1 "${SHARED_DIR}/mirror_registry_url"`
+if [ ! -f "${SHARED_DIR}/mirror_registry_url" ]; then
+    echo "File ${SHARED_DIR}/mirror_registry_url does not exist."
+    exit 1
+fi
+echo "MIRROR_REGISTRY_HOST: $MIRROR_REGISTRY_HOST"
+
+# target release
+target_release_image="${MIRROR_REGISTRY_HOST}/${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE#*/}"
+target_release_image_repo="${target_release_image%:*}"
+target_release_image_repo="${target_release_image_repo%@sha256*}"
+
+echo "target_release_image: $target_release_image"
+echo "target_release_image_repo: $target_release_image_repo"
+
+readable_version=$(oc adm release info "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}" --output=json | jq .metadata.version)
+echo "readable_version: $readable_version"
+
+# since ci-operator gives steps KUBECONFIG pointing to cluster under test under some circumstances,
+# unset KUBECONFIG to ensure this step always interact with the build farm.
+unset KUBECONFIG
+oc registry login
+
+# combine custom registry credential and default pull secret
+registry_cred=`head -n 1 "/var/run/vault/mirror-registry/registry_creds" | base64 -w 0`
+jq --argjson a "{\"${MIRROR_REGISTRY_HOST}\": {\"auth\": \"$registry_cred\"}}" '.auths |= . + $a' "${CLUSTER_PROFILE_DIR}/pull-secret" > "${new_pull_secret}"
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+BASTION_IP=$(<"${SHARED_DIR}/bastion_private_address")
+BASTION_SSH_USER=$(<"${SHARED_DIR}/bastion_ssh_user")
+
+# scp new_pull_secret credential to bastion host
+scp -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no "${new_pull_secret}" ${BASTION_SSH_USER}@${BASTION_IP}:/tmp/
+
+# mirror images in bastion host, which will increase mirror upload speed
+ssh -o UserKnownHostsFile=/dev/null -o IdentityFile="${SSH_PRIV_KEY_PATH}" -o StrictHostKeyChecking=no ${BASTION_SSH_USER}@${BASTION_IP} \
+oc adm release -a /tmp/new_pull_secret mirror --insecure=true \
```

Comment:
Because the `oc adm release` command is running on remote host, it is better to be quoted by "".

Reply:
Fixed